p4src/include/spgw.p4

```diff
@@ -92,87 +91,141 @@ control spgw_ingress(
         const default_action = nop();
     }
 
-#ifdef WITH_SPGW_PCC_GATING
-    action set_sdf_rule_id(sdf_rule_id_t id) {
-        fabric_meta.spgw.sdf_rule_id = id;
-    }
-
-    action set_pcc_rule_id(pcc_rule_id_t id) {
-        fabric_meta.spgw.pcc_rule_id = id;
+    action set_pdr_attributes(ctr_id_t ctr_id,
+                              far_id_t far_id) {
+        fabric_meta.spgw.pdr_hit = true;
+        fabric_meta.spgw.ctr_id = ctr_id;
+        fabric_meta.spgw.far_id = far_id;
     }
 
-    action set_pcc_info(pcc_gate_status_t gate_status) {
-        fabric_meta.spgw.pcc_gate_status = gate_status;
+    // These two tables scale well and cover the average case PDR
+    table downlink_pdr_lookup {
+        key = {
+            ipv4.dst_addr : exact @name("ue_addr");
+        }
+        actions = {
+            set_pdr_attributes;
+        }
     }
-
-    table sdf_rule_lookup {
+    table uplink_pdr_lookup {
         key = {
-            fabric_meta.spgw.direction   : exact @name("spgw_direction");
-            ipv4.src_addr                : ternary @name("ipv4_src");
-            ipv4.dst_addr                : ternary @name("ipv4_dst");
-            ipv4.protocol                : ternary @name("ip_proto");
-            fabric_meta.l4_sport         : ternary @name("l4_sport");
-            fabric_meta.l4_dport         : ternary @name("l4_dport");
+            // Is tunnel_ipv4_dst needed for Q2 target? Will it be variable?
```

Comment:
Yes. It will probably be static/unique for the same Tofino switch. Please double check with Ajay.